src/cloud/integrations/gitlab-integration.md

```diff
@@ -0,0 +1,173 @@
+   -  `--prune-branches`—An _optional_ parameter that instructs {{site.data.var.ece}} to delete branches that do not exist on the remote (`true` by default)
+
+   **Example**: Enable the GitLab integration for a private repository:
+
+   ```bash
+   magento-cloud integration:add --type gitlab --project=3txxjf32gtryos --token=qVUfeEn4ouze7A7JH --base-url=https://gitlab.com/ --server-project=my-agency/project-name --build-merge-requests=false --merge-requests-clone-parent-data=false --fetch-branches=true --prune-branches=true
+   ```
+
+1. When prompted, enter "Y" to add the integration.
+
+   ```terminal
+   Warning: adding a 'gitlab' integration will automatically synchronize code from the external Git repository.
+   This means it can overwrite all the code in your project.
+   Are you sure you want to continue? [y/N] y
+   ```
+
+1. Copy the **Hook URL** displayed by the return output.
+
+   ```terminal
+   Hook URL: https://eu-3.magento.cloud/api/projects/3txxjf32gtryos/integrations/eolmpfizzg9lu/hook
+   Created integration eolmpfizzg9lu (type: gitlab)
+   +----------------------------------+---------------------------------------------------------------------------------------+
+   | Property                         | Value                                                                                 |
+   +----------------------------------+---------------------------------------------------------------------------------------+
+   | id                               | eolmpfizzg9lu                                                                         |
+   | type                             | gitlab                                                                                |
+   | token                            | ******                                                                                |
+   | base_url                         | https://gitlab.com/                                                                   |
+   | project                          | my-agency/project-name                                                            |
+   | fetch_branches                   | true                                                                                  |
+   | prune_branches                   | true                                                                                 |
+   | build_merge_requests             | false                                                                                  |
+   | merge_requests_clone_parent_data | false                                                                                  |
+   | hook_url                         | https://eu-3.magento.cloud/api/projects/3txxjf32gtryos/integrations/eolmpfizzg9lu/hook |
+   +----------------------------------+---------------------------------------------------------------------------------------+
+   ```
+   {:.no-copy}
+
```

Comment:
The command template and field descriptions interrupts the flow of the procedure.  I think it will be easier to follow if the command explanation is in the overview, and the instructions provide the steps with the sample command.  For example:

```
## Enable the GitLab integration

Use the Magento Cloud CLI `integration` command to enable the GitLab integration, and get the Payload URL for the GitLab webhook to send updates from GitLab to your {{site.data.var.ece}} project.

```bash
magento-cloud integration:add --type=gitlab --project=<project-ID> --token=<your-GitLab-token> [--base-url=<GitLab-url> --server-project=<GitLab-project> --build-merge-requests={true|false} --merge-requests-clone-parent-data={true|false} --fetch-branches={true|false} --prune-branches={true|false}]
```

-  `<project-ID>`—Your {{site.data.var.ece}} project ID
-  `<your-GitLab-token>`—The personal access token you generated for GitLab
-  `--base-url`-URL of GitLab (https://gitlab.com/ if GitLab is used in its SaaS version)
-  `--server-project`-Project name in GitLab (part after the base url)
-  `--build-merge-requests`-—An _optional_ parameter that instructs {{site.data.var.ece}} to build a new environment for each and every merge requests (`true` by default)
-  `--merge-requests-clone-parent-data`-—An _optional_ parameter that instructs {{site.data.var.ece}} to clone the parent environment's data for merge requests (`true` by default)
-  `--fetch-branches`—An _optional_ parameter that causes {{site.data.var.ece}} to fetch all branches from the remote (as inactive environments) (`true` by default)
-  `--prune-branches`—An _optional_ parameter that instructs {{site.data.var.ece}} to delete branches that do not exist on the remote (`true` by default)

{:.bs-callout-warning}
The `magento-cloud integration` command overwrites _all_ code in your {{site.data.var.ece}} project with the code from your GitLab repository. This includes all branches, including the Production branch. This action happens instantly and cannot be undone. As a best practice, it is very important to clone all of your branches from your {{site.data.var.ece}} project and push them to your GitLab repository **before** adding the GitLab integration.

{:.procedure}
To enable the GitLab integration:

1. From the terminal, add the GitLab integration to your {{site.data.var.ece}} project:

   ```bash
   magento-cloud integration:add --type gitlab --project=3txxjf32gtryos --token=qVUfeEn4ouze7A7JH -- base-url=https://gitlab.com/ --server-project=my-agency/project-name --build-merge-requests=false --merge-requests-clone-parent-data=false --fetch-branches=true --prune-branches=true
   ```
   {:.no-copy}

1. When prompted, enter "Y" to add the integration.

   ```terminal
   Warning: adding a 'gitlab' integration will automatically synchronize code from the external Git repository.
   This means it can overwrite all the code in your project.
   Are you sure you want to continue? [y/N] y
   ```
   {:.no-copy}

1. Copy the **Hook URL** displayed in the return output.

   ```terminal
   Hook URL: https://eu-3.magento.cloud/api/projects/3txxjf32gtryos/integrations/eolmpfizzg9lu/hook
   Created integration eolmpfizzg9lu (type: gitlab)
   +----------------------------------+---------------------------------------------------------------------------------------+
   | Property                         | Value                                                                                 |
   +----------------------------------+---------------------------------------------------------------------------------------+
   | id                               | eolmpfizzg9lu                                                                         |
   | type                             | gitlab                                                                                |
   | token                            | ******                                                                                |
   | base_url                         | https://gitlab.com/                                                                   |
   | project                          | my-agency/project-name                                                            |
   | fetch_branches                   | true                                                                                  |
   | prune_branches                   | true                                                                                 |
   | build_merge_requests             | false                                                                                  |
   | merge_requests_clone_parent_data | false                                                                                  |
   | hook_url                         | https://eu-3.magento.cloud/api/projects/3txxjf32gtryos/integrations/eolmpfizzg9lu/hook |
   +----------------------------------+---------------------------------------------------------------------------------------+
   ```
   {:.no-copy}

```

Reply:
@Dnd-Mafer Appreciate your high quality contribution. A lot of review feedback, but that's partly because we need to clean up some of the existing older topics.

Reply:
Thank you ;)
I tried something for this chapter, let me know if it's OK for you.